### ISSUE TITLE
Stats: Modifying "Count logged in page views from" for custom roles does not work

### ISSUE THREAD

**Issue by richardmtl** [Automattic/jetpack#8161]
Originally reported here: https://wordpress.org/support/topic/setting-to-count-logged-in-page-views-from-customers-does-not-stick/

I was able to reproduce this on my test site. I have several custom roles:

<img width="331" alt="screen shot 2017-11-14 at 11 58 00 am" src="https://user-images.githubusercontent.com/3676359/32793143-1df2db9c-c933-11e7-94f6-3a058dd90a46.png">

I can click one of them and Activate it, and I get a green success message. However, when I refresh the page, the role is deactivated again.

**Comment by jeherve** [comment]
Does this happen in the old Jetpack > Site Stats > Configure page, or is the problem the same there as well?

**Comment by richardmtl** [comment]
Interesting question! So it turns out that some of the roles are already active there:

<img width="423" alt="screen shot on 2017-11-15 at 09-29-49 png" src="https://user-images.githubusercontent.com/3676359/32841150-9a5f5476-c9e7-11e7-82f8-cf15788cf518.png">

If I activate "Employer" it stays checked on that page; if I then go to the new Config screen, I see that it's still not active there. 

If I deactivate the 3 custom roles on the old page, and then activate it on the new page, it gets activated on the old page as well. However, if I refresh the new page, it still doesn't appear activated, though it remains activated on the old page on refresh. 

So, tl;dr is that the new page does not properly show the activation setting for custom roles (and custom roles only; I tested several default roles and they all worked as expected).

**Comment by aheckler** [comment]
As noted in 3526-gh-jpop-issues, this also affects changing which roles can view stats reports.

**Comment by chickenn00dle** [comment]
In reference to 1789238-zen 

The settings for which roles can view stats reports are actually changing, however custom users are still not able to view stats regardless of the change.

**Comment by Ilardo** [comment]
Hi, this also leads to situation that only administrators are capable of using the woocommerce mobile app to view shop statistics. No custom roles for example by woocommerce (shop manager) are capable of viewing the shop statistics via WOO mobile app or via wordpress.com only site administrators.

**Comment by jeherve** [comment]
@Ilardo Until this gets fixed I would recommend that you use the old Stats configuration screen (Jetpack > Site Stats > Configure) to change the stats access options. That should work.

**Comment by Ilardo** [comment]
> @Ilardo Until this gets fixed I would recommend that you use the old Stats configuration screen (Jetpack > Site Stats > Configure) to change the stats access options. That should work.

Hi, Thanks for your suggestion but i have tested this and this is only saving this option but there is no difference **in behaviour** for the custom role. So the old settings page save the settings but does not actually make any change in privileges.

**Comment by jeherve** [comment]
Thanks for letting us know, that's helpful!

**Comment by aheckler** [comment]
1924926-zen

**Comment by KokkieH** [comment]
https://wordpress.org/support/topic/site-stats-can-not-be-used-by-custom-roles/

**Comment by jeremeylduvall** [comment]
A VIP user bumped into this as well (96297 for my reference). I'm seeing references to the "old Configure" page in this thread and in others:

> Does this happen in the old Jetpack > Site Stats > Configure page, or is the problem the same there as well?

When I visit `example.com/wp-admin/admin.php?page=jetpack#/settings?term=Site%20Stats` in JP 5.2.2, I'm only presented with the new interface. Is there a way to still access the old interface or a workaround to use to enable stats for custom user roles while this is being worked on?

**Comment by jeherve** [comment]
> Is there a way to still access the old interface or a workaround to use to enable stats for custom user roles while this is being worked on?

We removed that old stats configuration page from Jetpack a few versions ago, so I am afraid there is no UI workaround right now.

**Comment by cafenoirdesign** [comment]
96297-zd-wordpressvip
98659-zd-wordpressvip
100987-zd-wordpressvip

**Comment by aitohamy** [comment]
The problem still there

**Comment by dericleeyy** [comment]
2997907-zen

**Comment by rawrndres** [comment]
This is still a current issue for VIP sites with custom roles.

VIP: (#114669-z)

**Comment by hitolonen** [comment]
Here are my findings:

**Default roles** 
- Wworks as expected. You can toggle active/deactive and everything runs smoothly.

**Custom roles**
- The initial toggle active works. It updates `stats_options`.
- On page refresh the toggle shows deactive even though `stats_options` has been updated.
- Changing the toggle setting doesn't update `stats_options`.

You can still update the setting with CLI.

Get the roles which have Stats permission, for example:

```
wp option pluck stats_options roles
array (
  0 => 'administrator',
  1 => 'custom-role',
)
```

And then delete custom-role with:

```
wp option patch delete stats_options roles 1
Success: Updated 'stats_options' option.
```

After some more testings, I've discovered that if you don't refresh the page, you can deactivate the role - this works after the initial page refresh too. Workflow:

Step 1: Go to `/wp-admin/admin.php?page=jetpack#/settings?term=Site Stats` and activate custom role.
=> You get settings updated message and option `stats_options` is updated.

Step 2: Hard refresh the page, try to deactivate the same custom role which shows as deactive.
=> You get settings updated message and `stats_options` stays the same.

Step 3: Do not refresh the page, and now with the custom role showing as active, deactivate that custom role.
=> You get settings updated message, the role now shows deactivated and option `stats_options` has been updated by removing the role.

**Comment by muffinpeace** [comment]
We've got a report of this on 4842500-zen with a user using WooCommerce custom-roles.

**Comment by raamdev** [comment]
We have another report of this issue on 146302-zd-wordpressvip.

**Comment by piotr-bajer** [comment]
Hi, is there any ETA for this ticket?

**Comment by coder-karen** [comment]
This issue has been fixed in https://github.com/Automattic/jetpack/pull/24887 - that will make its way into Jetpack 11.2 one month from now for self-hosted sites using Jetpack (so, the first Tuesday in August), or later this week for WoA sites. It can be used via the [Jetpack Beta Tester plugin](https://github.com/Automattic/jetpack-beta) in the interim though.
